apps/user-office-frontend-e2e/cypress/support/user.ts

```diff
@@ -1,234 +1,259 @@
+import { getE2EApi } from './utils';
+
+export type TestUserId = keyof typeof initialDBData.users;
+
+type DecodedTokenData = {
+  user: User;
+  currentRole: Role;
+  exp: number;
+  isInternalUser: boolean;
+};
+
+const extTokenStoreStfc = new Map<TestUserId, string>([
+  ['user1', 'user'],
+  ['user2', 'user'],
+  ['user3', 'user'],
+  ['officer', 'officer'],
+  ['placeholderUser', 'user'],
+  ['reviewer', 'user'],
+]);
+
+const getAndStoreFeaturesEnabled = (): Cypress.Chainable<GetFeaturesQuery> => {
+  const api = getE2EApi();
+  const request = api.getFeatures().then((resp) => {
+    const enabledFeatures = resp.features.filter((feat) => feat.isEnabled);
+    window.localStorage.setItem(
+      'enabledFeatures',
+      JSON.stringify(enabledFeatures)
+    );
+
+    return resp;
+  });
+
+  return cy.wrap(request);
+};
+
+function changeActiveRole(selectedRoleId: number) {
+  const token = window.localStorage.getItem('token');
+
+  if (!token) {
+    throw new Error('No logged in user');
+  }
+
+  const api = getE2EApi();
+  const request = api.selectRole({ selectedRoleId, token }).then((resp) => {
+    if (!resp.selectRole) {
+      return;
+    }
+
+    const { currentRole, user, exp, isInternalUser } = jwtDecode(
+      resp.selectRole
+    ) as DecodedTokenData;
+
+    window.localStorage.setItem('token', resp.selectRole);
+    window.localStorage.setItem(
+      'currentRole',
+      currentRole.shortCode.toUpperCase()
+    );
+    window.localStorage.setItem('expToken', `${exp}`);
+    window.localStorage.setItem('user', JSON.stringify(user));
+    window.localStorage.isInternalUser = isInternalUser;
+  });
+
+  cy.wrap(request);
+}
+
+const getUserIdFromIdOrCredentials = (
+  idOrCredentials: TestUserId | { email: string; password: string }
+) => {
+  const isCredentials = typeof idOrCredentials !== 'string';
+  if (isCredentials) {
+    const credentials = idOrCredentials;
+
+    const testUserId = Object.keys(initialDBData.users).find(
+      (key) =>
+        initialDBData.users[key as TestUserId].email === credentials.email &&
+        initialDBData.users[key as TestUserId].password === credentials.password
+    ) as TestUserId;
+
+    if (!testUserId) {
+      throw new Error(
+        `initialDBData object has no credentials for ${idOrCredentials.email}`
+      );
+    }
+
+    return testUserId;
+  } else {
+    return idOrCredentials as TestUserId;
+  }
+};
+
+const getCredentialsFromUserId = (testUserId: TestUserId) => {
+  const user = initialDBData.users[testUserId];
+
+  return {
+    email: user.email,
+    password: user.password,
+  };
+};
+
+const selectRole = async (token: string, selectedRoleId: number) => {
+  const api = getE2EApi();
+  const response = await api.selectRole({
+    token,
+    selectedRoleId,
+  });
+
+  return response.selectRole;
+};
+
+const getOauthExternalToken = async (testUserId: TestUserId) => {
+  const DEV_AUTH_SERVER_URL = 'http://localhost:5000';
```

Comment:
Could you make this environment variable? Since we have our local setup in the WSL localhost won't point to the auth server so will need to give our own URL to get the test working on local

Reply:
Will do it @TCMeldrum. Thanks for the good point.

Reply:
@TCMeldrum I added this `DEV_AUTH_SERVER_URL` in the cypress.json file as env variable. Let me know if this makes sense.

Reply:
Looks good thanks!